### DISCUSSION TITLE
How can I force neotree buffer to always be in normal mode

### DISCUSSION THREAD

**Discussion by airtonix** [nvim-neo-tree/neo-tree.nvim#667] [Q&A] [answered]
Yep, it's me again... your mouth breathing window licking mouse using heretic 👍🏻 🤝 

When I'm in `Insert Mode` and click into the neotree buffer... I'm still in `Insert Mode` which is basically useless to me and thus i waste roughly 5 seconds wondering why i can't open a folder or file...

How can i force interactions with a neotree buffer to always be in `Normal Mode` ?

**Comment by cseickel** [comment, marked as answer]
This will do it:
```lua
      local events = require("neo-tree.events")
      require("neo-tree").setup({
        event_handlers = {
          {
            event = events.NEO_TREE_BUFFER_ENTER,
            handler = function()
              vim.cmd("stopinsert")
            end,
          },
        },
      })

```
